src/api/java/baritone/api/behavior/look/IAimProcessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.api.behavior.look;
+
+import baritone.api.utils.Rotation;
+
+/**
+ * @author Brady
+ */
+public interface IAimProcessor {
+
+    /**
+     * Returns the actual rotation that will be used when the desired rotation is requested. This is not guaranteed to
+     * return the same value for a given input.
+     *
+     * @param desired The desired rotation to set
+     * @return The actual rotation
+     */
+    Rotation nextRotation(Rotation desired);
```

Comment:
I think this javadoc should explicitly say whether or not calling this function will advance the RNG by one tick, or not

Reply:
I think using one method to advance by one tick *or* peek the next tick depending on the instance is not good.
Can `IAimProcessor` maybe be split into one interface which can only be peeked and forked and a second interface which extends the first one by adding methods to tick it and to get a rotation and advance one tick simultaneously? The root processor would implement just the first interface, the forked ones the second interface.

Reply:
Agreed